### PR TITLE
Add AmbientImageLoader for Coil

### PR DESCRIPTION
**Pull request by chrisbanes**
Allows setting of a default `ImageLoader` across all `CoilImage`s.

### CODE REVIEW

**Review by manuelvicnt — APPROVED** [review]

File: coil/README.md [hunk 1]
```diff
@@ -93,6 +93,24 @@ CoilImage(
 
 Accompanist Coil supports GIFs through Coil's own GIF support. Follow the [setup instructions](https://coil-kt.github.io/coil/gifs/) and it should just work.
 
+## Custom ImageLoader
+
+If you wish to provide a default [`ImageLoader`](https://coil-kt.github.io/coil/image_loaders/) to use across all of your `CoilImage`
+calls, we provide the `AmbientImageLoader` ambient. You can see it like so:
```
Comment (nickbutcher) [inline coil/README.md:99]: Link to ambient docs? https://developer.android.com/reference/kotlin/androidx/compose/runtime/Ambient.html

File: coil/src/main/java/dev/chrisbanes/accompanist/coil/Coil.kt [hunk 2]
```diff
@@ -43,6 +44,21 @@ import dev.chrisbanes.accompanist.imageloading.ImageLoadState
 import dev.chrisbanes.accompanist.imageloading.MaterialLoadingImage
 import dev.chrisbanes.accompanist.imageloading.toPainter
 
+/**
+ * Ambient containing the preferred [ImageLoader] to use in [CoilImage].
+ */
+val AmbientImageLoader = staticAmbientOf<ImageLoader?> { null }
+
+object CoilImageConstants {
```
Comment (nickbutcher) [inline coil/src/main/java/dev/chrisbanes/accompanist/coil/Coil.kt:52]: Constants or defaults?

Comment (chrisbanes) [reply coil/src/main/java/dev/chrisbanes/accompanist/coil/Coil.kt:52]: I'm sticking with whatever Material does for now for consistency.

File: coil/README.md (hunk 1 above)
Comment (chrisbanes) [reply coil/README.md:99]: Added in https://github.com/chrisbanes/accompanist/commit/f2112d6270999e78c9748c83c7ead8d7ee0acc23

**Pull request merged**